lib/cbdocument.js

```diff
@@ -732,35 +732,58 @@ export default class CouchbaseDocument extends Document {
     const parts = options.populate.path.split('.');
     const part = parts[0];
 
+    let path;
+    let fullPath = false;
+    if (this.schema.hasRefPath(part)) {
+      path = part;
+    } else if (this.schema.hasRefPath(options.populate.path)) {
+      path = options.populate.path;
+      fullPath = true;
+    }
+
     // first part must be a ref path and cannot be a digit
-    if (!this.schema.hasRefPath(part) || /^\d+$/.test(part)) {
+    if (!path || /^\d+$/.test(path)) {
       return process.nextTick(() => {
         return fn(null, this, []);
       });
     }
 
-    let path = part;
-    const nextPart = parts[1];
-    let restIndex = 1;
-    // if next part is an array index append it to path and adjust
-    if (nextPart && /^\d+$/.test(nextPart)) {
-      path = part.concat('.', nextPart);
-      restIndex = 2;
-    }
+    const opts = clone(options);
+    let Model;
+    if (!fullPath) {
+      const nextPart = parts[1];
+      let restIndex = 1;
+      // if next part is an array index append it to path and adjust
+      if (nextPart && /^\d+$/.test(nextPart)) {
+        path = part.concat('.', nextPart);
+        restIndex = 2;
+      }
 
-    // create the rest of path
-    let rest;
-    if (parts.length > restIndex) {
-      rest = parts.slice(restIndex).join('.');
-    }
+      // create the rest of path
+      let rest;
+      if (parts.length > restIndex) {
+        rest = parts.slice(restIndex).join('.');
+      }
 
-    // get model
-    const Model = this._getRefModel(part);
-    if (!Model) {
-      console.warn('No model for path: %s', part);
-      return process.nextTick(() => {
-        return fn(null, this, []);
-      });
+      // get model
+      Model = this._getRefModel(part);
+      if (!Model) {
+        console.warn('No model for path: %s', part);
+        return process.nextTick(() => {
+          return fn(null, this, []);
+        });
+      }
+
+      // adjust the populate option for the rest
+      opts.populate.path = rest;
```

Comment:
Having both `options` and `opts` is confusing. Can we name them something a little more descriptive?